platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -1,49 +1,162 @@
 @RolesAllowed("dev")
 @Produces(MediaType.APPLICATION_JSON)
-@SecurityRequirement(name = "apiToken")
+@RequestScoped
 public class DeploymentEndpoint {
+  private static final Logger LOGGER = Logger.getLogger(DeploymentEndpoint.class);
   @Inject Mutiny.SessionFactory sf;
 
+  @Inject KubernetesClient client;
+
+  @GET
+  @Path("{deploymentName}")
+  public Uni<ObjectMeta> getStream(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getK8Deployment(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/logs")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Uni<String> getLogs(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getDeploymentLogs(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/watchLogs")
+  @Produces(MediaType.SERVER_SENT_EVENTS)
+  public Response watchLogs(
+      @PathParam("deploymentName") String deploymentName,
+      @Context Sse sse,
+      @Context SseEventSink eventSink)
+      throws IOException {
+    LogWatch lw = watchDeploymentLogs(deploymentName).watchLog();
+    InputStream is = lw.getOutput();
+    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(is));
+    String line;
+    while ((line = bufferedReader.readLine()) != null) {
+      final OutboundSseEvent sseEvent = sse.newEventBuilder().data(line).build();
+
+      eventSink.send(sseEvent);
+    }
+    is.close();
+    lw.close();
+    return Response.ok().build();
+  }
+
   @GET
-  public Uni<List<DeploymentEntity>> getDeployments() {
-    return sf.withSession(
-        session ->
-            session.createQuery("from DeploymentEntity", DeploymentEntity.class).getResultList());
+  public Uni<ListMeta> getDeployments() {
+    return Uni.createFrom().item(getK8Deployments());
   }
 
   @POST
-  @RequestBody
-  @Consumes(MediaType.APPLICATION_JSON)
-  public Uni<Deployment> createDeployment(Deployment deployment) throws JsonProcessingException {
-    DeploymentEntity de = new DeploymentEntity(deployment.spec());
-    return sf.withTransaction(
-        (session, transaction) -> session.merge(de).replaceWith(() -> deployment));
+  @Path("{pipelineUuid}")
+  public Uni<String> createDeployment(@PathParam("pipelineUuid") UUID pipelineId) {
+    return apply(pipelineId);
   }
 
   @DELETE
-  public Uni<DeploymentSpec> deleteDeployment(DeploymentSpec spec) {
-    CriteriaBuilder cb = sf.getCriteriaBuilder();
-    CriteriaDelete<DeploymentEntity> delete = cb.createCriteriaDelete(DeploymentEntity.class);
-    Root<DeploymentEntity> root = delete.from(DeploymentEntity.class);
-    delete.where(cb.equal(root.get("spec"), spec));
+  @Path("{deploymentName}")
+  public Uni<Response> deleteDeployment(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(deleteK8Deployment(deploymentName));
+  }
+
+  @PUT
+  @Path("{pipelineUuid}")
+  public Uni<String> updateDeployment(@PathParam("pipelineUuid") UUID pipelineId) {
```

Comment:
At some point, one pipeline could have multiple deployments. When updating a deployment, we should reference it by the deployment's uuid.

Reply:
reference the new deployment uuid and change pipelineuuid from `@PathParam` to `@QueryParam`